lms/djangoapps/courseware/courses.py

```diff
@@ -415,3 +415,24 @@ def get_studio_url(course, page):
     if is_studio_course and is_mongo_course:
         studio_link = get_cms_course_link(course, page)
     return studio_link
+
+
+def get_problems_in_section(section):
+    """
+    This returns list of problems in a section.
+    """
+
+    problem_descriptors = []
+    exam_key = UsageKey.from_string(section)
+    # it will be a Mongo performance boost, if you pass in a depth=3 argument here
+    # as it will optimize round trips to the database to fetch all children for the current node
+    section = modulestore().get_item(exam_key, depth=3)
+
+    # iterate over section, sub-section,  unit
+    for subsection in section.get_children():
+        for vertical in subsection.get_children():
+            for unit in vertical.get_children():
+                if unit.location.category == 'problem':
```

Comment:
Also, can I suggest you use the term 'component' for 'unit', becuase normally 'unit' = 'vertical', but here you are enumerating over all children of a vertical, so the right nomenclature to use here is component or even module.